neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -21,9 +22,32 @@ public class SetVersion { public VersionPayload Version; }
         public class SetVerack { }
         public class SetFilter { public BloomFilter Filter; }
 
+        private class LimitedSet<T>
+        {
+            private int maxCapacity;
+            private OrderedDictionary orderedDictionary;
+
+            public LimitedSet(int maxCapacity = 100)
+            {
+                if (maxCapacity <= 0) throw new ArgumentOutOfRangeException(nameof(maxCapacity));
+
+                this.maxCapacity = maxCapacity;
+                this.orderedDictionary = new OrderedDictionary(maxCapacity);
+            }
+
+            public bool Add(T item)
+            {
+                if (orderedDictionary.Contains(item)) return false;
+                if (orderedDictionary.Count >= maxCapacity)
+                    orderedDictionary.RemoveAt(0);
+                orderedDictionary.Add(item, null);
```

Comment:
Since you are adding null as the value, you could just be using a sorted structure that doesn’t have a value like `SortedSet`. However as mentioned above it should really use a structure that evicts least recently accessed items.

Reply:
Actually sortedset doesn’t help removing in an order so I’ll retract this comment.